scripts/Enlighten.py

```diff
@@ -78,16 +81,21 @@ def parse_args(self, argv):
     #
     def create_parser(self):
         parser = argparse.ArgumentParser(description="acquire from specified device, display line graph")
-        parser.add_argument("--log-level",          type=str, default="info",       help="logging level",    choices=['debug', 'info', 'warning', 'error', 'critical'])
-        parser.add_argument("--log-append",         action="store_true",            help="append to existing logfile")
-        parser.add_argument("--logfile",            type=str,                       help="Explicit path for the logfile")
-        parser.add_argument("--max-memory-growth",  type=int, default=0,            help="Automatically exit after this percent memory growth (0 for never, 100 = doubling)")
-        parser.add_argument("--run-sec",            type=int, default=0,            help="Automatically exit after this many seconds (0 for never)")
-        parser.add_argument("--serial-number",      type=str,                       help="only connect to specified serial number")
-        parser.add_argument("--set-all-dfu",        action="store_true",            help="set spectrometers to DFU mode as soon as they connect")
-        parser.add_argument("--stylesheet-path",    type=str,                       help="Path to CSS directory")
-        parser.add_argument("--headless",           action="store_true",            help="Run Enlighten without GUI")
-        parser.add_argument("--plugin",             type=str,                       help="Plugin name")
+
+        # This code was like a spreadsheet when I found it, leaning into that
+        # use :set nowrap in vim
+        # TODO: everything should have a default -- do not rely on empty args
+        #                  | parameter name       | type    | default             | action             | choices                                                  | help |
+        parser.add_argument("--log-level",         type=str, default="info",                            choices=['debug', 'info', 'warning', 'error', 'critical'], help="logging level")
+        parser.add_argument("--log-append",        type=str, default="LIMIT",                      choices=["False", "True", "LIMIT"],                   help="append to existing logfile")
```

Comment:
Not quite, no.

Here's how it could go:

## Scenario A, arbitrarily long sessions

1) Using my existing Enlighten with "--log-append=false" by default, I may create some arbitrarily large log file. Enlighten always cleared these files _only at start up time_, so as long as the most recent one was a fairly log-heavy long session, the file may get quite large.

2) Using a newly installed Enlighten with "--log-append=limit", the log will be instantly[^1] trimmed to the limit size driven by Wasatch.PY (currently 20mb). Like previous versions, the log can still grow to an arbitrary size if there is a long log-heavy session. Trimming happens only at start up time.

3) The next start up will do the same thing, it will trim, but the left over 20mb seems to slow down the startup.

4) This will happen indefinitely, until the user manually deletes the log file, because once it reaches a 20mb limit, it will always be at least that big.

## Scenario B, several normal sessions that add up

1) Since the log is appending, even if the sessions are individually small they will add up.

2) Eventually the log file will reach capacity and all subsequent start ups will be slowed down.

## Conclusion

The slow boot would eventually affect everyone. My working theory is that it's the 20mb that is causing a slowdown somewhere, and that any size beyond that does not introduce any additional slowdown. That theory is supported by the fact that I observed the same slow start up on Enlighten main and Wasatch on 501e78a3. Still it would not hurt to further unit test the File Resize Program, just in case, I didn't intend for sizes 10gb+ when I wrote it.

The slow boot should be fixed one way or another before merging. So this is now a draft.

## Action Step

Either 

A) We reduce the limit size such that the boot time is acceptable. For example, if 10 sec is okay, then we can figure out the corresponding size (approx 2mb). This is super easy, but it kind of blows--2mb is like 42 seconds of historical logs.

B) (preferred) We eliminate the cause of slowdown pre-merge.

One hacky way to do it would be to keep the enlighten.log from the previous session in enlighten.log.temp until Enlighten is done booting up, then moving it in-place. I don't recommend we actually do that, but thinking that way may shed some light towards an actual solution.

[^1]: It looked like it was able to resize a 50gb file instantly. This can be verified by further unit testing of the [File resize program](https://github.com/WasatchPhotonics/Wasatch.PY/blob/f2a28fc07442ae9c6c3ebb21883381d2d68f8019/wasatch/utils.py#L596).